Mathlib/RingTheory/LinearDisjoint.lean

```diff
@@ -0,0 +1,604 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.LinearAlgebra.LinearDisjoint
+import Mathlib.LinearAlgebra.TensorProduct.Subalgebra
+import Mathlib.LinearAlgebra.Dimension.FreeAndStrongRankCondition
+import Mathlib.LinearAlgebra.FreeModule.StrongRankCondition
+import Mathlib.Algebra.Algebra.Subalgebra.MulOpposite
+import Mathlib.Algebra.Algebra.Subalgebra.Rank
+import Mathlib.RingTheory.IntegralClosure.Algebra.Defs
+import Mathlib.RingTheory.IntegralClosure.IsIntegral.Basic
+
+/-!
+
+# Linearly disjoint of subalgebras
+
+This file contains basics about the linearly disjoint of subalgebras.
+
+## Main definitions
+
+- `Subalgebra.LinearDisjoint`: two subalgebras are linearly disjoint, if they are
+  linearly disjoint as submodules (`Submodule.LinearDisjoint`).
+
+- `Subalgebra.LinearDisjoint.mulMap`: if two subalgebras `A` and `B` of `S / R` are
+  linearly disjoint, then there is `A ⊗[R] B ≃ₐ[R] A ⊔ B` induced by multiplication in `S`.
+
+## Main results
+
+### Equivalent characterization of linearly disjointness
+
+- `Subalgebra.LinearDisjoint.linearIndependent_left_op_of_flat`:
+  `Subalgebra.LinearDisjoint.linearIndependent_left_of_flat_of_commute`:
+  `Subalgebra.LinearDisjoint.linearIndependent_left_of_flat`:
+  if `A` and `B` are linearly disjoint, if `B` is a flat `R`-module, then for any family of
+  `R`-linearly independent elements of `A`, they are also `B`-linearly independent
+  (in the opposite ring for the first one).
```

Comment:
Why not drop the `op` and `of_commute` lemmas here -- this is Main results, not necessarily all results. Then you can just explain one lemma clearly instead of adding brackets to try and explain three lemmas in one sentence (and similarly for the next three).

Reply:
OK.